src/main/java/com/exasol/adapter/response/converter/ResponseJsonConverter.java

```diff
@@ -0,0 +1,138 @@
+ */
+public final class ResponseJsonConverter {
+    private static final String SCALAR_FUNCTION_PREFIX = "FN_";
+    private static final String PREDICATE_PREFIX = "FN_PRED_";
+    private static final String AGGREGATE_FUNCTION_PREFIX = "FN_AGG_";
+    private static final String LITERAL_PREFIX = "LITERAL_";
+    private static final String SCHEMA_METADATA = "schemaMetadata";
+    private static final ResponseJsonConverter responseJsonConverter = new ResponseJsonConverter();
+
+    private ResponseJsonConverter() {
+    }
+
+    /**
+     * Returns instance of {@link ResponseJsonConverter} singleton class
+     *
+     * @return {@link ResponseJsonConverter} instance
+     */
+    public static ResponseJsonConverter getInstance() {
+        return responseJsonConverter;
+    }
+
+    /**
+     * Converts drop virtual schema response into a JSON format
+     *
+     * @param dropResponse instance
+     * @return string representation of a JSON Object
+     */
+    @SuppressWarnings("squid:S1172")
+    public String convertDropVirtualSchemaResponse(final DropVirtualSchemaResponse dropResponse) {
+        return Json.createObjectBuilder() //
+              .add("type", "dropVirtualSchema") //
+              .build() //
+              .toString();
+    }
+
+    /**
+     * Converts create virtual schema response into a JSON format
+     *
+     * @param createResponse instance
+     * @return string representation of a JSON Object
+     */
+    public String convertCreateVirtualSchemaResponse(final CreateVirtualSchemaResponse createResponse) {
+        return Json.createObjectBuilder() //
+              .add("type", "createVirtualSchema") //
+              .add(SCHEMA_METADATA, SchemaMetadataSerializer.serialize(createResponse.getSchemaMetadata())) //
+              .build() //
+              .toString();
+    }
+
+    /**
+     * Converts push down response into a JSON format
+     *
+     * @param pushDownResponse instance
+     * @return string representation of a JSON Object
+     */
+    public String convertPushDownResponse(final PushDownResponse pushDownResponse) {
+        return Json.createObjectBuilder() //
+              .add("type", "pushdown") //
+              .add("sql", pushDownResponse.getPushDownSql()) //
+              .build() //
+              .toString();
+    }
+
+    /**
+     * Converts get capabilities response into a JSON format
+     *
+     * @param getCapabilitiesResponse instance
+     * @return string representation of a JSON Object
+     */
+    public String convertGetCapabilitiesResponse(final GetCapabilitiesResponse getCapabilitiesResponse) {
+        final JsonObjectBuilder builder = Json.createObjectBuilder().add("type", "getCapabilities");
+        final JsonArrayBuilder arrayBuilder = Json.createArrayBuilder();
+        final Capabilities capabilities = getCapabilitiesResponse.getCapabilities();
+        for (final MainCapability capability : capabilities.getMainCapabilities()) {
```

Comment:
Extract this and the next four for loops into a separate method each.

Reply:
done